pkg/sbproxy/sbproxy.go

```diff
@@ -106,14 +110,29 @@ func New(ctx context.Context, cancel context.CancelFunc, settings *Settings, pla
 	log.AddHook(&logging.ErrorLocationHook{})
 
 	util.HandleInterrupts(ctx, cancel)
+	filters := []web.Filter{
+		&filters.Logging{},
+	}
+	authnSettings := settings.Authentication
+	if len(authnSettings.User) != 0 && len(authnSettings.Password) != 0 {
```

Comment:
General question: Shouldn't we always expect to have authentication settings provided?

And then just introduce a Validate method on the new authentication settings to use before constructing the basic and oidc filters?

Reply:
This way we will make it mandatory for whoever uses the OSS agent to have to configure authentication even if he doesnt really want to use the config api